src/services/redisSessionStorage.ts

```diff
@@ -313,17 +322,138 @@ export default class RedisSessionStorage extends RedisStorageUtils
+      this.redisClient.srem(blockedUserSetKey, fiscalCode, (err, response) =>
+        resolve(
+          this.falsyResponseToError(
+            this.integerReply(err, response, 1),
+            new Error(
+              "Unexpected response from redis client deleting blockedUserKey"
+            )
+          )
+        )
+      );
+    });
+  }
+
+  /**
+   * Check if a user is blocked
+   *
+   * @param fiscalCode id of the user
+   *
+   * @returns a promise with either an error or a boolean indicating if the user is blocked
+   */
   public async isBlockedUser(
     fiscalCode: FiscalCode
   ): Promise<Either<Error, boolean>> {
-    return this.sismemberTask("BLOCKED-USERS", fiscalCode)
+    return this.sismemberTask(blockedUserSetKey, fiscalCode)
       .bimap(
         err => new Error(`Error accessing blocked users collection: ${err}`),
         result => result === 1
       )
       .run();
   }
 
+  /**
+   * Delete all user session data
+   * @param fiscalCode
+   */
+  public async delUserAllSessions(
+    fiscalCode: FiscalCode
+  ): Promise<Either<Error, boolean>> {
+    const sessionsOrError = await this.readSessionInfoKeys(fiscalCode);
+
+    const delSingleSession = (
+      token: SessionToken
+    ): Promise<Either<Error, boolean>> =>
+      this.loadSessionBySessionToken(token)
+        .then(e => {
+          const user: User = e.getOrElseL(err => {
+            throw err;
+          });
+          return this.del(user.session_token, user.wallet_token);
+        })
+        .catch(_ => {
+          // if I didn't find a user by it's token, I assume there's nothing about that user, so its data is deleted already
+          return _ === sessionNotFoundError
+            ? right<Error, boolean>(true)
+            : left(_);
+        });
+
+    const delEverySession = (sessionTokens: readonly SessionToken[]) =>
+      array
+        .sequence(taskEither)<Error, boolean>(
+          sessionTokens.map(sessionInfoKey =>
+            fromEither<Error, SessionToken>(
+              SessionToken.decode(sessionInfoKey).mapLeft(
+                _ => new Error("Error decoding token")
+              )
+            ).chain<boolean>((token: SessionToken) =>
+              tryCatch(() => delSingleSession(token), toError).chain(fromEither)
+            )
+          )
+        )
+        .map(() => true);
+
+    return fromEither(sessionsOrError)
+      .foldTaskEither<Error, boolean>(
+        _ => (_ === sessionNotFoundError ? taskEither.of(true) : fromLeft(_)),
+        sessionInfoKeys =>
+          delEverySession(
+            sessionInfoKeys.map(
+              sessionInfoKey =>
+                sessionInfoKey.replace(sessionInfoKeyPrefix, "") as SessionToken
+            )
+          )
+      )
+      .chain(_ =>
+        tryCatch(() => this.delSessionInfoKeys(fiscalCode), toError).chain(
+          fromEither
+        )
+      )
+      .run();
+  }
+
+  private delSessionInfoKeys(
```

Comment:
This method delete `UserSessionsSet`

Reply:
Would `delSessionsSet` be preferable?

Reply:
The point is this method is simmetric with
```
  private readSessionInfoKeys(
    fiscalCode: FiscalCode
  ): Promise<Either<Error, ReadonlyArray<string>>> {
    return new Promise<Either<Error, ReadonlyArray<string>>>(resolve => {
      this.redisClient.smembers(
        `${userSessionsSetKeyPrefix}${fiscalCode}`,
        (err, response) => resolve(this.arrayStringReply(err, response))
      );
    });
  }
```

Which, as well, reads from `userSessionsSet`. 

So I think the correct point is: `delSessionInfoKeys` correctly deletes `userSessionsSet`, but it also has to iterate every `sessionInfo` and delete each one.